src/Microsoft.ML.Data/Scorers/FeatureContributionCalculation.cs

```diff
@@ -0,0 +1,449 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Text;
+using Microsoft.ML.Data;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.CommandLine;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.Internal.Internallearn;
+using Microsoft.ML.Runtime.Internal.Utilities;
+using Microsoft.ML.Runtime.Model;
+using Microsoft.ML.Runtime.Numeric;
+
+[assembly: LoadableClass(typeof(IDataScorerTransform), typeof(BindableMapper), typeof(BindableMapper.Arguments),
+    typeof(SignatureDataScorer), "Feature Contribution Transform", "fct", "FeatureContributionCalculationTransform", MetadataUtils.Const.ScoreColumnKind.FeatureContribution)]
+
+[assembly: LoadableClass(typeof(ISchemaBindableMapper), typeof(BindableMapper), typeof(BindableMapper.Arguments),
+    typeof(SignatureBindableMapper), "Feature Contribution Mapper", "fct", MetadataUtils.Const.ScoreColumnKind.FeatureContribution)]
+
+[assembly: LoadableClass(typeof(BindableMapper), typeof(BindableMapper), null, typeof(SignatureLoadModel),
+    "Feature Contribution Mapper", BindableMapper.MapperLoaderSignature)]
+
+namespace Microsoft.ML.Runtime.Data
+{
+    /// <summary>
+    /// Holds the definition of the getters for the FeatureContribution column. It also contains the generic mapper that is used to score the Predictor.
+    /// This is only used by the command line API.
+    /// </summary>
+    internal sealed class BindableMapper : ISchemaBindableMapper, ICanSaveModel, IPredictor
```

Comment:
>BindableMapper [](start = 26, length = 14)

This name is rather strangely generic. Was this intended to be nested class?

Reply:
I see, so I should make this into a class.

---
In reply to: [243036387](https://github.com/dotnet/machinelearning/pull/1847#discussion_r243036387) [](ancestors = 243036387)